test/utils/Arrays.test.js

```diff
@@ -0,0 +1,81 @@
+const ArraysImpl = artifacts.require('ArraysImpl');
+
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('Arrays', function () {
+  context('Odd number of elements', function () {
+    const ODD_ELEMENTS_ARRAY = range(128, 256);
+
+    beforeEach(async function () {
+      this.arrays = await ArraysImpl.new(ODD_ELEMENTS_ARRAY);
+    });
+
+    it('should return correct index for the basic case', async function () {
+      (await this.arrays.findUpperBound(139)).should.be.bignumber.equal(11);
+    });
+
+    it('should return `index 0` for the first element', async function () {
+      (await this.arrays.findUpperBound(128)).should.be.bignumber.equal(0);
+    });
+
+    it('should return `last index` for the last element', async function () {
+      (await this.arrays.findUpperBound(256)).should.be.bignumber.equal(128);
+    });
+
+    it('should return `last index + 1` for the element over the upper boundary', async function () {
+      (await this.arrays.findUpperBound(512)).should.be.bignumber.equal(129);
```

Comment:
The fact that the array is 129 elements long, and the first element is 128, may be a bit confusing. Consider using e.g. a shorter array.